rdkPlugins/Networking/source/Netfilter.cpp

```diff
@@ -679,257 +778,159 @@ bool Netfilter::ruleInList(const std::string &rule,
+ *  @return always returns true.
  */
-bool Netfilter::insertRules(const RuleSet &ruleSet, const int ipVersion)
+bool Netfilter::insertRules(RuleSet &ruleSet, const int ipVersion)
 {
     AI_LOG_FN_ENTRY();
 
-    // get the existing iptables rules
-    RuleSet existing = getRuleSet(ipVersion);
+    // get reference to the correct cache
+    RuleSets &ruleCache = (ipVersion == AF_INET) ? mIpv4RuleCache : mIpv6RuleCache;
 
-    // create a rule set from entries in the supplied argument that are not
-    // not in the existing rules
-    RuleSet actual;
-
-    // check if we already have any of the rules
-    for (const std::pair<const TableType, std::list<std::string>> &newRule : ruleSet)
+    for (auto &it : ruleSet)
     {
-        const TableType& table = newRule.first;
-        const std::list<std::string>& tableRules = newRule.second;
-
-        // get the existing table
-        const std::list<std::string> &existingRules = existing[table];
-
-        // add the rule to the actual list if it exists
-        for (const std::string &rule : tableRules)
+        // find rules table from cache
+        auto cacheRuleset = ruleCache.insertRuleSet.find(it.first);
+        if (cacheRuleset == ruleCache.insertRuleSet.end())
         {
-            if (!ruleInList(rule, existingRules))
-            {
-                actual[table].emplace_back(rule);
-            }
+            // the table doesn't exist, so add it as a new table
+            ruleCache.insertRuleSet.emplace(it);
+        }
+        else
+        {
+            // table exists, merge new rules to the end of it
+            cacheRuleset->second.merge(it.second);
         }
-    }
-
-    if (actual.empty())
-    {
-        AI_LOG_INFO("all iptables rules are already set");
-        AI_LOG_FN_EXIT();
-        return true;
-    }
-
-    // finally apply the rules
-    bool success = applyRuleSet(Operation::Insert, actual, ipVersion);
-    if (!success)
-    {
-        AI_LOG_ERROR("failed to insert all iptables rules");
     }
 
     AI_LOG_FN_EXIT();
-    return success;
+    return true;
 }
 
 // -----------------------------------------------------------------------------
 /**
- *  @brief Uses the iptables-restore tool to atomically delete a set of rules
+ *  @brief Adds delete rules to the internal rule caches to be written later.
  *
- *  The function pipes the rules in @a ruleSet into the iptables-restore tool,
- *  prefixed with the -D option.
- *
- *  This is equivalent to running the following for all the rules
- *     iptables -t <table> -D <rule>
+ *  These rules will be prefixed with "-D" to signal deleting from iptables.
  *
  *  @param[in]  ruleSet         The ruleset to apply.
  *  @param[in]  ipVersion       iptables version to use.
  *
- *  @return true on success, false on failure.
+ *  @return always returns true.
  */
-bool Netfilter::deleteRules(const RuleSet &ruleSet, const int ipVersion)
+bool Netfilter::deleteRules(RuleSet &ruleSet, const int ipVersion)
```

Comment:
This name is very confusing as "deleteRules" as this functon is adding more elements, not removing them is more like "addDeleteRules". Also those all "deleteRules", "appendRules", "insertRules" looks like they have so similar logic that if written correctly they could be one function, but I think that it is not worth changing as you wrote this one already. So in my opinion just change naming to "addDeleteRules" and other probably too, to be consistent.